### PR TITLE
feat: protected route redirect overwrite per page

### PR DESCRIPTION
**Pull request by LouisHaftmann**
Adds a `navigateUnauthenticatedTo` option to `definePageMeta()` to allow for overwriting the default behavior of redirecting to the login page

### CODE REVIEW

**Review by BracketJohn — COMMENTED** [review]
Sorry for taking so long to get around to this - was quite busy with my day job.

I like the feature - would just be good to also add some documentation here, same as in other PR! 

If you add aome we should be ready to go, thanks!

File: src/runtime/middleware/auth.ts [hunk 1]
```diff
@@ -3,8 +3,23 @@ import { navigateToAuthPages, determineCallbackUrl } from '../utils/url'
 import { useAuth } from '#imports'
 
 type MiddlewareMeta = boolean | {
-  unauthenticatedOnly: true,
+  /** Whether to only allow unauthenticated users to access this page.
```
Comment (BracketJohn) [inline src/runtime/middleware/auth.ts:6]: CI is failing, should mostly be fixable by running `pnpm lint --fix` (:

**Review by BracketJohn — COMMENTED** [review]
As in other PR of yours: Please move your docs-changes to the `v0.6` docs, thanks!

**Review by zoey-kaiser — DISMISSED** [review]
LGTM!

File: src/runtime/middleware/auth.ts [hunk 2]
```diff
@@ -60,7 +75,5 @@ export default defineNuxtRouteMiddleware((to) => {
     const signInOptions: Parameters<typeof signIn>[1] = { error: 'SessionRequired', callbackUrl: determineCallbackUrl(authConfig, () => to.path) }
     // @ts-ignore This is valid for a backend-type of `authjs`, where sign-in accepts a provider as a first argument
     return signIn(undefined, signInOptions) as ReturnType<typeof navigateToAuthPages>
-  } else {
-    return navigateTo(authConfig.provider.pages.login)
-  }
+  } else if (typeof metaAuth === 'object' && metaAuth.navigateUnauthenticatedTo) { return navigateTo(metaAuth.navigateUnauthenticatedTo) } else { return navigateTo(authConfig.provider.pages.login) }
```
Comment (valiafetisov) [inline src/runtime/middleware/auth.ts:78]: Let's increase readability 🙂 
```suggestion
  } else if (typeof metaAuth === 'object' && metaAuth.navigateUnauthenticatedTo) {
    return navigateTo(metaAuth.navigateUnauthenticatedTo)
  } else {
    return navigateTo(authConfig.provider.pages.login)
  }
```

File: src/runtime/middleware/auth.ts [hunk 3]
```diff
@@ -3,8 +3,23 @@ import { navigateToAuthPages, determineCallbackUrl } from '../utils/url'
...
 
 type MiddlewareMeta = boolean | {
-  unauthenticatedOnly: true,
+  /** Whether to only allow unauthenticated users to access this page.
+   *
+   * Authenticated users will be redirected to `/` or the route defined in `navigateAuthenticatedTo`
+   *
+   * @default undefined
+   */
+  unauthenticatedOnly?: boolean,
```
Comment (valiafetisov) [inline src/runtime/middleware/auth.ts:12]: You did made this value to be optional, but you didn't set the default value anywhere. I suggest to keep `true` to be the default to not change previous expected behaviour

**Review by zoey-kaiser — APPROVED** [review]

**Pull request merged**